src/app/printerprofile.service.ts

```diff
@@ -0,0 +1,69 @@
+import { HttpClient, HttpErrorResponse } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Router } from "@angular/router";
+import { Subscription } from "rxjs";
+
+import { ConfigService } from "./config/config.service";
+import { NotificationService } from "./notification/notification.service";
+import { PrinterService } from "./printer.service";
+
+import { OctoprintPrinterProfileAPI } from "./octoprint-api/printerProfileAPI";
+
+@Injectable({
+  providedIn: "root",
+})
+export class PrinterProfileService {
+  private httpGETRequest: Subscription;
+  
+  public constructor(
+    private http: HttpClient,
+    private configService: ConfigService,
+    private notificationService: NotificationService,
+    private printerStatusService: PrinterService,
+    private router: Router
+  ) { }
+
+  public getDefaultPrinterProfile(): Promise<OctoprintPrinterProfileAPI> {
+    return new Promise((resolve): void => {
+      if (this.httpGETRequest) {
+        this.httpGETRequest.unsubscribe();
+      }
+      this.httpGETRequest = this.http
+        .get(this.configService.getURL("printerprofiles/_default"), this.configService.getHTTPHeaders())
+        .subscribe(
+          (data: OctoprintPrinterProfileAPI): void => {
+            const printerProfile: OctoprintPrinterProfileAPI = data;
+            resolve(printerProfile);
+          },
+          (error: HttpErrorResponse): void => {
+            const printerProfile: OctoprintPrinterProfileAPI = {
+              name: "_default",
+              model: "unknown",
+              axes: {
+                x: { inverted: false },
+                y: { inverted: false },
+                z: { inverted: false }
+              }
+            };
```

Comment:
I would probably remove this and instead reject the observable (just catch it in control.component and do nothing there). Since you already added the default profile in the control.component returning nothing here might be better. This will either keep the default profile or, if a prior request was successful, keep the last retrieved printer profile, which should be better than the default. Let me know if you want to work on that, or whether I should fix that :).